src/core/server/elasticsearch/elasticsearch_config.ts

```diff
@@ -39,7 +40,23 @@ export const config = {
       defaultValue: 'http://localhost:9200',
     }),
     preserveHost: schema.boolean({ defaultValue: true }),
-    username: schema.maybe(schema.string()),
+    username: schema.conditional(
+      schema.contextRef('dev'),
```

Comment:
Instead of using `schema.contextRef('dev')`, how do you feel about using `schema.contextRef('dist')`?

When running Kibana with `node ./scripts/kibana.js --dev` from source, we're in "dev mode": dev is true, prod is false, dist is false.

When running Kibana with `node ./scripts/kibana.js` from source, we're in "production mode": dev is false, prod is true, dist is false.

And when running Kibana from a built artifact: dev is false, prod is true, dist is true.


If nothing else is preventing us from using `schema.contextRef('dist')`, we'd ensure that `elastic` is used in even less situations.

Reply:
I'm fine with that, I guess we are making the assumption that people aren't actually running Kibana from source in production? Or if they are, we are accepting that we may break their installation in a minor release?

Reply:
> I'm fine with that, I guess we are making the assumption that people aren't actually running Kibana from source in production

I'd sure hope not!

> Or if they are, we are accepting that we may break their installation in a minor release?

I think the likelihood of this impacting users is low enough that we're tolerant of the potential impact.